packages/smart-wallet/src/smartWallet.js

```diff
@@ -990,21 +991,23 @@ export const prepareSmartWallet = (baggage, shared) => {
             // await so that any errors are caught and handled below
             await watchOfferOutcomes(watcher, seatRef);
           } catch (err) {
-            facets.helper.logWalletError('OFFER ERROR:', err);
+            // This block only runs if the block above fails during one vat incarnation.
+            facets.helper.logWalletError('IMMEDIATE OFFER ERROR:', err);
 
-            // Notify the user
+            // Update status to observers
             if (err.upgradeMessage === 'vat upgraded') {
```

Comment:
@Chris-Hibbert I just checked again, and this is the only non-test check manually using `upgradeMessage` rather than the `isUpgradeDisconnection`. I'm satisfied enough at our analysis that this is harmless to request it in this PR.

Reply:
@turadg, since you're still making changes, would you mind updating this?
```suggestion
            if (isUpgradeDisconnection(err)) {
```

Reply:
oh yeah, I meant to.